### ISSUE TITLE
IARU HF Championship

### ISSUE THREAD

**Issue by K6OK** [w7sst/MorseRunner#158]
1. [IARU HF Championship](https://www.arrl.org/iaru-hf-world-championship)
2. Rules: https://contests.arrl.org/ContestRules/IARU-HF-Rules.pdf
3. Call history file: https://n1mmwp.hamdocs.com/mmfiles/iaru_2022-007-txt/

Contest has dual exchange: 
amateur stations send 5NN and ITU zone
IARU country HQ stations typically send 5NN and exchange of variable length (typically 2 to 5 chars)

It is impossible to know ahead of time which HQ's will be on the air.  Must rely on prior year's history file.
The file above only has HQ stations.   K6OK can create a call history that merges call history from Cabrillos with HQ stations to create a more representative total call history.

**Comment by w7sst** [comment]
Thanks for adding this. Turns out I had already added as Issue #63. I marked the original as a duplicate to keep this one. I like the idea of the combined history file.

**Comment by f6fvy** [comment]
HQ exchanges are not limited to 4 chars. They can be from 2 chars to 6 chars. Some HQ don't even use the same abbreviation along the years. Several different callsigns can use the same exchange (Italian or Japanese HQ or even IARU members).

**Comment by K6OK** [comment]
@f6fvy -- Thank you for catching that.  Original post above has been corrected.

**Comment by w7sst** [comment]
@f6fvy @K6OK - I've been coding the IARU HF Contest and have it limping. I have a question...

For regular stations sending their ITU-Zone numbers, how often will these Zone numbers be sent using cut numbers? Example zone 1 as A, zone 10 as AT, zone 19 as 1N, etc.? I'm trying to figure out coding rules for verifying the exchange.

In particular, the society codes AC, R1, R2, and R3 are problematic. When sending, will operators shorten R1 to RA? How does the receiving station know this is R1 and not RA?

I'm just looking for some guidance on how to coding both the sending logic (where we cut numbers) and the log checking logic (where we expand the cut numbers) for comparison.

Thank you, Mike

**Comment by f6fvy** [comment]
Regarding the cut numbers for the ITU zones, I would say 5 or 10% at most.

The cut numbers are never used for the HQ and IARU members exchanges.

**Comment by w7sst** [comment]
@K6OK @f6fvy - I have the IARU HF Contest limping...

> K6OK can create a call history that merges call history from Cabrillos with HQ stations to create a more representative total call history.

Jim, I'll take you up on your offer to merge some Cabrillos file(s) into the IARU call history file. You are right, there are only HQ stations in the call history file and no other users with regular Zone entries.

I also notice that some of the HQ station's callsigns are not in our DXCC file. We knew this when we decided to use the DXCC file that we would not have 100% coverage. This only affects the status bar printout of Continent/Entity. For example, C4HQ (Cuba HQ) is not in the list.

I still have to work on the cut number behaviors.

[Here](https://nextcloud.brashlerphoto.com/s/XeYFFD56m5nWPyq) is a version with the IARU HF Contest added.

**Comment by K6OK** [comment]
@w7sst -- Done!  Pull request posted.

**Comment by w7sst** [comment]
I am starting to work on this issue again. @f6fvy provided a C++ version of a CTY Country file reader. I am in the process of porting this to Pascal. This will provide robust Country lookups for most, if not all, callsigns. I have attached this issue to the v1.84 milestone which will release in the Apr-Jun timeframe.

@scotthibbs @k6ok @ct7aup

**Comment by w7sst** [comment]
I finally got back to looking at and working on this IARU Contest. I've been away for sometime now due to some medical issues here at home.

Last spring, I had completed a port of @f6fvy 's CTY Country file reading to Pascal. I had it basically working. However, this past month, I ended up adding all the remaining regular expressions to our DXCC file (`ARRL.LIST`). It turns out that with these additional regular expressions, the DXCC file will correctly answer most callsign lookups to find the call's corresponding Entity. While it does not return the correct Zone, it does return a suitable Entity string. This entity string is _only_ used in the status message bar above the callsign field to show user's continent/entity (e.g. `EU/France`). All zones being sent by MRCE are taken from the call history file.

So, my conclusion at this point is to release the IARU HF Contest using the updated DXCC file, and not finish the CTY parser.

Lastly, I'm thinking of changing the pending 1.83.1 bug-fix release and call it 1.84 and include IARU HF Contest since it is basically finished at this point.

Thoughts?

Thank you,
Mike

@K6OK @f6fvy @ct7aup @scotthibbs

**Comment by scotthibbs** [comment]
I like new contests, I can not lie! Hihi.

1.84 it is!

**Comment by f6fvy** [comment]
Go for 1.84 ;-) Thanks !

**Comment by w7sst** [comment]
A copy of the 1.84 pre-release build is available [here](https://1drv.ms/u/s!AiN4lELeOz01q0LYuPt75uEGNMEM?e=rryom5).

**Comment by K6OK** [comment]
Mike -- with 1.84-rc1 I made about 100 QSOs with a mix of callsigns in different continents.   Tested both Run and Single Calls.  It's not an issue per se, but I never did hear one of the HQ stations.  The history file is heavily weighted with regular calls and the random draw never picked an HQ.  To test the HQ calls I made a temporary history file that had a 50/50 mix of regular and HQ calls, just make sure there's no problems with the HQ exchanges.  No problems to report.

73 Jim K6OK

@Mike @f6fvy @scotthibbs
